dashboard/app/views/levels/editors/fields/_csa_fields.html.haml

```diff
@@ -16,6 +16,43 @@
     - maze_value = @level.serialized_maze ? JSON.pretty_generate(@level.serialized_maze) : ''
     = f.text_area :serialized_maze, value: maze_value, rows: 20, class: 'input-block-level'
 
+-# This element exists instead of the _encrypted_examples element in order to facilitate level urls
+-# instead of project urls in the example solutions. This is needed because Java Lab has mini-app
+-# types, such as neighborhood, which are highly dependent on their level definition and do not have
+-# a standalone type. However, we can't use the same method used by Maze or other puzzle level
```

Comment:
I am mostly just curious, but what is the background on Java Lab having mini-app types? Is there a reason these are not each their own app type?

Reply:
Most of the code is shared between the Java Lab mini-apps. i.e. the messaging & communication between Javabuilder (backend) and Java Lab, the editor & editor display, the console IO, and the general IDE. The only thing that's different between them is the visualization element and some of the signals they are set up to receive from Javabuilder.